### PR TITLE
FAI-11611: GitHub Issues Streams

### PR DESCRIPTION
**Pull request by matiaslcoulougian**
## Description
Issues source stream and converter + some linting and cleaning

ToDo:
- [x] Add tests
- [x] Add permissions

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: destinations/airbyte-faros-destination/src/converters/github/faros_issues.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        model: 'tms_Label',
+        record: {tms_Label: {name: label}},
+      });
+    }
+    return res;
+  }
+
+  async onProcessingComplete(): Promise<ReadonlyArray<DestinationRecord>> {
+    return [
+      ...this.convertUsers(),
```
Comment (dbruno21) [inline destinations/airbyte-faros-destination/src/converters/github/faros_issues.ts:167]: Is this needed? (writing vcs_User records)

File: destinations/airbyte-faros-destination/src/converters/github/faros_issues.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    if (!assignee.login) return null;
+    return {
+      assignee,
+      createdAt: assignment.createdAt,
+      isAssigned: assignment.type === 'AssignedEvent',
+    };
+  }
+
+  private convertLabels(): ReadonlyArray<DestinationRecord> {
+    const res: DestinationRecord[] = [];
```
Comment (dbruno21) [inline destinations/airbyte-faros-destination/src/converters/github/faros_issues.ts:155]: If this is how we end up writing tms_Label records, I don't see why we should do something different for vcs_Label records. 

Right now: 
- Labels stream writes vcs_Label
- PR stream writes vcs_PullRequestLabel

If we're doing it this way for issues, for consistency, PR stream should write vcs_Label and then Labels stream is kind of useless.

File: destinations/airbyte-faros-destination/src/converters/github/faros_issues.ts (hunk 1 above)
Comment (matiaslcoulougian) [reply destinations/airbyte-faros-destination/src/converters/github/faros_issues.ts:167]: good question, maybe not. In the feed we were writing only tms_User...

File: destinations/airbyte-faros-destination/src/converters/github/faros_issues.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        statusChangelog,
+        createdAt,
+        updatedAt: Utils.toDate(issue.updatedAt),
+        creator: issue.author?.login
+          ? {uid: issue.author.login, source}
+          : undefined,
+      },
+    });
+
+    // TODO: If tasks get transferred between repos or projects, delete previous relationship
```
Comment (matiaslcoulougian) [inline destinations/airbyte-faros-destination/src/converters/github/faros_issues.ts:114]: Should we implement this? Brought this TODO comment from feeds. This would imply adding a query to Faros graph to check if there is any current project relationship != current project to delete it. cc @chalenge

Comment (chalenge) [reply destinations/airbyte-faros-destination/src/converters/github/faros_issues.ts:114]: Okay to keep as is. We can fix later if we find use for these

File: destinations/airbyte-faros-destination/src/converters/github/faros_issues.ts (hunk 2 above)
Comment (chalenge) [reply destinations/airbyte-faros-destination/src/converters/github/faros_issues.ts:155]: How easy would it be to pass the flag that pulls issues to the converter, then write both `vcs` and `tms` if we pulling issues?

File: destinations/airbyte-faros-destination/test/resources/faros_github/catalog.json [hunk 4]
```diff
@@ -95,6 +95,12 @@
         "name": "mytestsource__github__faros_projects"
       },
       "destination_sync_mode": "overwrite"
+    },
+    {
+      "stream": {
+        "name": "mytestsource__github__faros_issues"
+      },
+      "destination_sync_mode": "overwrite"
```
Comment (chalenge) [inline destinations/airbyte-faros-destination/test/resources/faros_github/catalog.json:103]: think this should be `append`

File: faros-airbyte-common/resources/github/queries/assignee-fields-fragment.gql [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        login
+        name
+        html_url: url
+    }
+    ... on User {
+        login
+        name
+        html_url: url
+    }
+}
```
Comment (chalenge) [inline faros-airbyte-common/resources/github/queries/assignee-fields-fragment.gql:21]: nit: newline

File: faros-airbyte-common/resources/github/queries/issues-query.gql [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+              assignee {
+                ...assigneeFields
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
```
Comment (chalenge) [inline faros-airbyte-common/resources/github/queries/issues-query.gql:63]: nit: newline

File: faros-airbyte-common/resources/github/queries/issues-query.gql [hunk 7]
```diff
@@ -0,0 +1,63 @@
+query issues($owner: String!, $repo: String!, $page_size: Int = 25, $cursor: String) {
```
Comment (chalenge) [inline faros-airbyte-common/resources/github/queries/issues-query.gql:1]: why is the `page_size` hard-coded?

File: faros-airbyte-common/resources/github/queries/pull-requests-query.gql [hunk 8]
```diff
@@ -81,10 +81,10 @@ query pullRequests($owner: String!, $repo: String!, $cursor: String, $page_size:
             login
           }
         }
-        ...Labels
```
Comment (chalenge) [inline faros-airbyte-common/resources/github/queries/pull-requests-query.gql:84]: Can we add these unrelated changes to their own PRs to reduce noise

File: destinations/airbyte-faros-destination/src/converters/github/faros_issues.ts (hunk 2 above)
Comment (matiaslcoulougian) [reply destinations/airbyte-faros-destination/src/converters/github/faros_issues.ts:155]: Thought about something similar. Should not be that complex

File: faros-airbyte-common/resources/github/queries/issues-query.gql (hunk 7 above)
Comment (matiaslcoulougian) [reply faros-airbyte-common/resources/github/queries/issues-query.gql:1]: We are setting this default value in case the parameter is not set for all queries. But we are always adding it to the params using the pageSize constant, with `DEFAULT_PAGE_SIZE` = 100 by default.

Comment (matiaslcoulougian) [reply faros-airbyte-common/resources/github/queries/issues-query.gql:1]: Also, I'm seeing in some queries we set = 25 and others = 100.

Comment (matiaslcoulougian) [reply faros-airbyte-common/resources/github/queries/issues-query.gql:1]: But as I said, we are always setting it in the query parameters. Maybe we should remove them.

**Review by chalenge — APPROVED** [review]
lgtm

**Review by dbruno21 — APPROVED** [review]
